quinn-runtime/Cargo.toml

```diff
@@ -0,0 +1,21 @@
+[package]
+name = "quinn-runtime"
```

Comment:
What's the benefit of introducing a new crate, rather than defining this inside `quinn`?

Reply:
As you noticed yourself after writing the initial comment, we can not define it in `quinn` because `quinn-udp` relies on it. Defining it in `quinn-udp` would be weird because it contains things that have nothing to do with udp.
I could change it to put only the AsyncWrappedUdpSocket inside `quinn-udp` and the rest into `quinn`.
But also this seperate crate helps understand the dependencies structure better. The runtime crate depends on tokio and/or async-std/io. quinn and quinn-udp just depend on the runtime crate. Makes it harder to accidentally introduce direct tokio calls into the rest of quinn-udp again.

Reply:
> I could change it to put only the AsyncWrappedUdpSocket inside quinn-udp and the rest into quinn.

This sounds broadly reasonable to me. Alternatively, it should be possible to refactor quinn-udp to be entirely runtime-agnostic, without requiring traits at all, just primitive I/O operations that the caller is responsible for calling at the right times.

> Makes it harder to accidentally introduce direct tokio calls into the rest of quinn-udp again.

We can address this with CI if necessary. Extra crates means extra maintenance effort, which we must keep to a minimum.

Reply:
Implemented in the other PR